### PR TITLE
2223 inbound pd schema fix

### PR DESCRIPTION
**Pull request by jonahkaye**
Ticket: #[2223](https://github.com/metriport/metriport-internal/issues/1766)

### Description

- inbound xcpd schema error fix

### Testing

- Local
  - [x] on local offending xmls
- Staging
  - [ ] staging

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/schema.ts [hunk 1]
```diff
@@ -40,10 +40,12 @@ export const QueryByParameterSchema = z.object({
...
-    livingSubjectName: z.object({
-      value: schemaOrArray(nameSchema),
-      semanticsText: z.literal("LivingSubject.name"),
-    }),
+    livingSubjectName: schemaOrArray(
+      z.object({
+        value: nameSchema,
+        semanticsText: z.literal("LivingSubject.name").optional(),
+      })
+    ),
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/schema.ts:48]: `{
    "code": "invalid_type",
    "expected": "object",
    "received": "array",
    "path": [
        "Envelope",
        "Body",
        "PRPA_IN201305UV02",
        "controlActProcess",
        "queryByParameter",
        "parameterList",
        "livingSubjectName"
    ],
    "message": "Expected object, received array"
}`

Fix for this schema error. LivingSubjectName can be an array while value is only allowed to be a single object
https://profiles.ihe.net/ITI/TF/Volume2/ITI-55.html

File: packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/create/iti55-envelope.ts [hunk 2]
```diff
@@ -172,18 +172,20 @@ function createSoapBodyContent({
...
+                      [`${prefix}family`]: name.family,
+                      ...name.given?.reduce((acc: { [key: string]: string }, givenName: string) => {
+                        acc[`${prefix}given`] = givenName;
+                        return acc;
+                      }, {}),
+                    },
+                    [`${prefix}semanticsText`]: "LivingSubject.name",
+                  },
+                }))
+              : []),
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/create/iti55-envelope.ts:188]: update request creation accordingly

File: packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts [hunk 3]
```diff
@@ -84,7 +84,7 @@ export async function processInboundXcpdRequest(
     return inboundRequest;
   } catch (error) {
     const msg = "Failed to parse ITI-55 request";
-    log(`${msg}: Error - ${errorToString(error)}`);
+    log(`${msg}: Error - ${errorToString(error)}, JsonObj: ${JSON.stringify(jsonObj)}`);
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts:87]: better logging for when we error

File: packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/schema.ts [hunk 4]
```diff
@@ -40,14 +40,16 @@ export const QueryByParameterSchema = z.object({
...
     patientAddress: z
       .object({
         value: schemaOrArrayOrEmpty(addressSchema).optional(),
-        semanticsText: z.literal("Patient.addr"),
+        semanticsText: z.literal("Patient.addr").optional(),
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/schema.ts:52]: `{
    "received": "Patient.address",
    "code": "invalid_literal",
    "expected": "Patient.addr",
    "path": [
        "Envelope",
        "Body",
        "PRPA_IN201305UV02",
        "controlActProcess",
        "queryByParameter",
        "parameterList",
        "patientAddress",
        "semanticsText"
    ],
    "message": "Invalid literal value, expected \"Patient.addr\""
}`

seems some people dont provide. no need to make this req

File: packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts [hunk 5]
```diff
@@ -15,9 +15,9 @@ export function transformIti55RequestToPatientResource(
   const queryParams =
     iti55Request.Envelope.Body.PRPA_IN201305UV02.controlActProcess.queryByParameter.parameterList;
 
-  const name = toArray(queryParams.livingSubjectName.value).map(name => ({
-    family: extractText(name.family),
-    given: toArray(name.given).map(extractText),
+  const name = toArray(queryParams.livingSubjectName).map(value => ({
```
Comment (RamilGaripov) [inline packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts:18]: nit: `value.value`. Maybe rename to `.map(name =>...`

File: packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts (hunk 3 above)
Comment (RamilGaripov) [reply packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/process/xcpd-request.ts:87]: nit: `JsonObj` is not very descriptive here. What does it represent?

**Review by RamilGaripov — APPROVED** [review]
Approved on the condition that the nits are addressed

**Pull request merged**